api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCcrCohortQueries.java

```diff
@@ -0,0 +1,3014 @@
+   *       de registo de mais que uma data “PCR (data da colheita)” durante o período de reporte,
+   *       será considerada a primeira ocorrência.
+   * </ul>
+   *
+   * @return {@link CohortDefinition}
+   */
+  public CohortDefinition getChildrenFirstPcrCollectedUnder2MonthsofAge() {
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+    cd.setName("1º PCR colhido < 2 meses de idade");
+    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("location", "Health Facility", Location.class));
+
+    cd.addSearch("firstConsultation", map(getPatients1stConsultation(), mapping));
+    cd.addSearch("received1stPcr", map(getChildrenFirstPcr(), mapping));
+    cd.addSearch("pcrbellow2monthsOfAge", map(getInfantAgeAtPcr(false, 2), mapping));
+
+    cd.setCompositionString("firstConsultation AND received1stPcr AND pcrbellow2monthsOfAge");
+    return cd;
+  }
+
+  /**
+   * CCR-FR19
+   *
+   * <p><b>Indicador 13 - </b>1º PCR colhido ≥ 2 meses de idade
+   *
+   * <p>O sistema irá produzir o Indicador 13 “Total de 1º PCR colhido ≥ 2 meses de idade”, da
+   * seguinte forma:
+   *
+   * <ul>
+   *   <li>Incluindo todas as crianças que tiveram a 1ª consulta durante o período de reporte (CCR-
+   *       FR7)
+   *   <li>Filtrando as que tiveram o registo de “PCR (data da colheita)”, na “Ficha de Seguimento
+   *       de CCR”, sendo essa data “PCR (data da colheita)” durante o período de reporte (“PCR
+   *       (data da colheita)” >= “Data Início” e <= “Data Fim”) e com idade >= 2 meses nesta data
+   *       (“PCR (data da colheita” – “Data de Nascimento” >= 2 meses). Nota: no caso de existência
+   *       de registo de mais que uma data “PCR (data da colheita)” durante o período de reporte,
+   *       será considerada a primeira ocorrência.
+   * </ul>
+   *
+   * @return {@link CohortDefinition}
+   */
+  public CohortDefinition getChildrenFirstPcrCollectedAbove2MonthsofAge() {
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+    cd.setName("1º PCR colhido >= 2 meses de idade");
+    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("location", "Health Facility", Location.class));
+
+    cd.addSearch("firstConsultation", map(getPatients1stConsultation(), mapping));
+    cd.addSearch("received1stPcr", map(getChildrenFirstPcr(), mapping));
+    cd.addSearch("pcrabove2monthsOfAge", map(getInfantAgeAtPcr(true, 2), mapping));
+
+    cd.setCompositionString("firstConsultation AND received1stPcr AND pcrabove2monthsOfAge");
+    return cd;
+  }
+
+  public CohortDefinition getGeneralCcrQuery(Concept questionConcept, List<Integer> answerConcept) {
+    SqlCohortDefinition cd = new SqlCohortDefinition();
+    cd.setName("Ficha Seguimento CCR Query");
```

Comment:
consider receiving a list of concepts instead of integers

Reply:
Resolved